pkg/combustion/rpm.go

```diff
@@ -32,13 +35,38 @@ func configureRPMs(ctx *image.Context) ([]string, error) {
 
 	zap.L().Info("Configuring RPM component...")
 
-	var rpmDir string
+	var localRPMConfig *image.LocalRPMConfig
 	if isComponentConfigured(ctx, userRPMsDir) {
-		rpmDir = generateComponentPath(ctx, userRPMsDir)
+		rpmDir := generateComponentPath(ctx, userRPMsDir)
+		localRPMConfig = &image.LocalRPMConfig{
+			RPMPath: rpmDir,
+		}
+
+		gpgPath := filepath.Join(rpmDir, userGPGsDir)
+		_, err := os.Stat(gpgPath)
+
+		packages := ctx.ImageDefinition.OperatingSystem.Packages
+		switch {
+		case err == nil:
+			if !packages.NoGPGCheck {
+				localRPMConfig.GPGKeysPath = gpgPath
+			} else {
+				log.Auditf("WARNING: Found %s directory, when GPG validation is disabled", gpgPath)
```

Comment:
Isn't this where we want to fail due to a conflict? The audit and warning logs I mentioned earlier referred to notifying the user that setting `NoGPGCheck: true` is in general not advised.

Reply:
I didn't keep EIB's failure convention in mind. This has been fixed, if we have a GPG dir, but GPG validation is disabled we fail instead of giving a warning. Furthermore, I added a warning log when using the `noGPGChecks: true` parameter.